### DISCUSSION TITLE
Additional download for offline cdn?

### DISCUSSION THREAD

**Discussion by ceciiiron** [alpinejs/alpine#3735] [1. Help] [answered]
I love this framework. However I have a new project that needs to be hosted offline which means I need the offline alpine.js file. 

I tried downloading the js file from https://cdn.jsdelivr.net/npm/alpinejs@3.13.0/dist/cdn.min.js, however, `x-data` does not read my defined data but `Alpine.store()` and `template`, `x-if` works so I think I need to download some additional js files but I do not know where to locate it.

I am not using a web compiler so I cannot npm install this framework.

I've already read discussion https://github.com/alpinejs/alpine/discussions/2338 but like I said, some alpine codes are not working

Update:
This code works
```html
<div x-data="{firstName: 'third'}">
     <p>Hello world <span x-text="firstName"></span></p>
</div>
```

However, when I try to separate the x-data something like
```html
<div x-data="myApp()">
     <p>Hello world <span x-text="firstName"></span></p>
</div>
```
```js
function myApp() {
     return {
          firstName: "third",
     }
}
```
The code above is what I usually do if i am using the online cdn. It doesn't work if I am using the downloaded js from the online cdn.

**Comment by ekwoka** [comment, marked as answer]
No, everything works the same.

The cdn one is the same one npm installs. It's not really different.

What I'd suspect is that you're not putting `defer` on the script.

You also don't need a `web compiler`. You can still use npm to get the package.

**Reply by ceciiiron** [reply]
Thank you! I did missed adding `defer` on my script. Now everything works great.
